browser/data-browser/src/views/OntologyPage/PropertyDatatypePicker.tsx

```diff
@@ -16,29 +23,34 @@ export function PropertyDatatypePicker({
   resource,
   disabled,
 }: PropertyDatatypePickerProps): JSX.Element {
-  const [_, setAllowsOnly] = useArray(resource, urls.properties.allowsOnly, {
+  const [, setAllowsOnly] = useArray(resource, core.properties.allowsOnly, {
+    commit: true,
+  });
+  const [, setClassType] = useString(resource, core.properties.classtype, {
     commit: true,
   });
 
-  const removeAllowsOnlyForNonResourceArray = (type: string) => {
-    if (
-      type === urls.datatypes.resourceArray ||
-      type === urls.datatypes.atomicUrl
-    ) {
-      return;
-    }
+  const isResourceLike = (datatype: string) => {
+    return (
+      datatype === Datatype.ATOMIC_URL || datatype === Datatype.RESOURCEARRAY
+    );
+  };
 
-    setAllowsOnly(undefined);
+  const handleChange = (datatype: string) => {
```

Comment:
The name `handleChange` is a bit too generic. It doesn't actually handle setting the value that changes as that is done inside the `AtomicSelectInput`. Calling it something like `removeTypeSpecificProps` would be better in this case

Reply:
`clearInapplicableProps()`?

Reply:
@Polleps changed the name -- better now?

Reply:
Yes perfect!